secrets/README.md

```diff
@@ -0,0 +1,226 @@
+# Secret Management
+
+The `secrets` package provides a unified way to handle secrets within configuration files for Prometheus and its ecosystem components. It allows secrets to be specified inline, loaded from files, or fetched from other sources through a pluggable provider mechanism.
+
+## Concepts
+
+The package is built around a few core concepts:
+
+  * `SecretField`: A type used in configuration structs to represent a field that holds a secret. It handles the logic for unmarshaling from different secret sources, and the API for accessing secrets.
+  * `ProviderConfig`: An interface for configuring a secret provider. It includes methods to create a new `Provider` instance and to clone the configuration.
+  * `ProviderConfigId`: An interface for uniquely identifying a `ProviderConfig` instance.
+  * `Provider`: An interface for fetching secrets from a specific source (e.g., inline string, file on disk). The package comes with built-in providers, and new ones can be registered.
+  * `Manager`: A component that discovers all `SecretField` instances within a configuration struct, manages their lifecycle, and handles periodic refreshing of secrets.
```

Comment:
Can we put those into comments? Especially for library, unlikely anyone will notice README. In go it's either `doc.go` or code structure for godocs.